### DISCUSSION TITLE
URGENT: Deployed to Forge server and now website broken, getting Attribute [stripeWebhooks] does not exist.

### DISCUSSION THREAD

**Discussion by hondaman900** [spatie/laravel-stripe-webhooks#150] [Q&A]
Everything is working well on my local dev site. Webhooks working well with Stripe CLI. I pushed to my Forge server and my whole site is now broken. In the Forge deployment log it's saying

```
In RouteRegistrar.php line 92:
                                              
  Attribute [stripeWebhooks] does not exist.
```
In my `web.php` routing file I have

`Route::stripeWebhooks('stripe-webhook');`

which works fine in dev.

I found two others in the discussion group for this package who had this same issue and neither found a resolution. One reverted to installing Cashier and I don't have that option.

What is different in the deployed environment that would trip this up? Is something not getting pushed to the live server that would cause this, and if so, why?

**Comment by freekmurze** [comment]
Since it works at your end locally and our testsuite passes, I'm going to assume the root of the issue is something with your production environment. 

As a quick fix you could try to manually register the routes that the package tries to register using the `stripeWebhooks` macro.

**Reply by hondaman900** [reply]
Thank you. I'm starting to think it's a deeper issue on Forge. I don't see the `laravel-stripe-webhooks` package being installed as it should and trying to do so manually via CL fails, citing composer out-of-memory errors when it tries to install the package. I suspect the automated deployment scripts encountered the same composer error and failed the deployment. I have a tech support call into Forge to see if they can resolve this or increase the Composer memory allocation for this package.

FWIW, here's the error:
```
Warning from https://repo.packagist.org: Support for Composer 1 is deprecated and some packages will not be available. You should upgrade to Composer 2. See https://blog.packagist.com/deprecating-composer-1-support/
Info from https://repo.packagist.org: #StandWithUkraine
Using version ^2.2 for spatie/laravel-stripe-webhooks
./composer.json has been updated
Loading composer repositories with package information
Warning from https://repo.packagist.org: Support for Composer 1 is deprecated and some packages will not be available. You should upgrade to Composer 2. See https://blog.packagist.com/deprecating-composer-1-support/
Info from https://repo.packagist.org: #StandWithUkraine
Updating dependencies (including require-dev)
PHP Fatal error:  Allowed memory size of 1610612736 bytes exhausted (tried to allocate 67108872 bytes) in phar:///usr/local/bin/composer/src/Composer/DependencyResolver/RuleSet.php on line 83

Fatal error: Allowed memory size of 1610612736 bytes exhausted (tried to allocate 67108872 bytes) in phar:///usr/local/bin/composer/src/Composer/DependencyResolver/RuleSet.php on line 83

Check https://getcomposer.org/doc/articles/troubleshooting.md#memory-limit-errors for more info on how to handle out of memory errors.
```

Apologies for posting an issue for what is turning out to be a Forge server issue, and many thanks for this excellent package.

**Reply by hondaman900** [reply]
I ran composer update on my local dev to create a new `composer.lock` which I committed to my live site, in the hopes that Composer on the live server wouldn't need so much memory for a `composer install` since it wouldn't have to figure out dependencies with the new `composer.lock`. That ran withotu error on Forge, but `laravel-stripe-webhooks` didn't get installed on the live end. I then searched through the `composer.lock` file, and there's no mention of `laravel-stripe-webhooks`. My other Spatie packages are there. 

Why is `laravel-stripe-webhooks` being excluded from my `composer.lock` and `composer.json` files?
